lib/hooks/use-order-quote-polling.ts

```diff
@@ -0,0 +1,132 @@
+import type { OrderQuote } from "@tscircuit/fake-snippets/schema"
+import { registryKy } from "lib/utils/get-registry-ky"
+import { useEffect, useRef, useState } from "react"
+import { useCallback } from "react"
+
+type PollingState = {
+  status: "idle" | "loading" | "success" | "error"
+  data: OrderQuote | null
+  error: Error | null
+}
+
+export const useOrderQuotePolling = (packageReleaseId: string) => {
+  const [state, setState] = useState<PollingState>({
+    status: "idle",
+    data: null,
+    error: null,
+  })
+  const abortControllerRef = useRef<AbortController | null>(null)
+
+  const createOrderQuotes = useCallback(async () => {
+    const { order_quote_id } = await registryKy
+      .post("order_quotes/create", {
+        json: {
+          package_release_id: packageReleaseId,
+          vendor_name: "jlcpcb", // TODO: Remove this once we have more vendor
+        },
+      })
+      .json<{ order_quote_id: string }>()
+    return order_quote_id
```

Comment:
why is it `createOrderQuotes` if it returns a single order quote? We will have multiple order quotes when we do `/order_quotes/create_for_all_vendors`, but then you should have the function signature match the eventual output (API/signature is more important to maintain than internal implementations, which can be temporary)

to match the eventual output you would return an array of ids